### PR TITLE
ARROW-6016: [Python] Fix get_library_dirs() when Arrow installed as a system package

### PR DESCRIPTION
**Pull request by pitrou**
Presumably this should fix the issue, though I can't check.

### CODE REVIEW

**Comment by mbrobbel** [comment]
This fixed my issue 👍

**Comment by pitrou** [comment]
Thanks for the quick answer @mbrobbel :-)

**Comment by pitrou** [comment]
@xhochy @kszucs Any concerns?

File: python/pyarrow/__init__.py [hunk 1]
```diff
@@ -262,8 +262,11 @@ def append_library_dir(library_dir):
         if _has_pkg_config(pkgname):
             library_dir = _read_pkg_config_variable(pkgname,
                                                     ["--libs-only-L"])
-            assert library_dir.startswith("-L")
-            append_library_dir(library_dir[2:])
+            # pkg-config output could be empty if Arrow is installed
+            # as a system package.
+            if library_dir:
+                assert library_dir.startswith("-L")
```
Comment (xhochy) [inline python/pyarrow/__init__.py]: This is not test code, so we should rather raise than `assert`?

Otherwise LGTM

**Comment by pitrou** [comment]
+1. Will merge.

**Pull request closed without merging**